packages/mattermost-redux/src/selectors/entities/roles.ts

```diff
@@ -152,29 +140,42 @@ export const getMyTeamPermissions: (state: GlobalState, options: PermissionsOpti
     },
 );
 
-export const getMyChannelPermissions: (state: GlobalState, options: PermissionsOptions) => Set<string> = createSelector(
-    'getMyChannelPermissions',
-    getMyChannelRoles,
-    getRoles,
-    getMyTeamPermissions,
-    (state, options: PermissionsOptions) => options.channel,
```

Comment:
Taking a look at the old code, I don't think the `options` object would've actually caused any extra recalculation if these used a proper selector factory. The result function is only re-ran when one of the first n-1 selectors returns a new result. Since `options.channel` is just a string in this case, the result function wouldn't recalculate as long as this was called with an identical object twice in a row.